src/test/samVisualize/graphGeneration/graphGeneration.test.ts

```diff
@@ -0,0 +1,106 @@
+/*!
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import * as assert from 'assert'
+import { generateGraphFromYaml } from '../../../samVisualize/graphGeneration/cfnTemplateGraphGenerator'
+import { Node, Link } from '../../../samVisualize/graphGeneration/graph'
+import { join } from 'path'
+import { readdirSync, readFileSync } from 'fs'
+import { getProjectDir } from '../../testUtil'
+
+const TestInputDirectory = join(
+    getProjectDir(),
+    'testFixtures',
+    'workspaceFolder',
+    'samVisualize-test-data',
+    'testYamlTemplates'
+)
+const TestExpectedOutputDirectory = join(
+    getProjectDir(),
+    'testFixtures',
+    'workspaceFolder',
+    'samVisualize-test-data',
+    'expectedGraphObjects'
+)
+
+let inputFiles: Array<string>
+let expectedOutputFiles: Array<string>
+
+describe('samVisualize Graph Generation from YAML', async function () {
```

Comment:
Is it worth testing any error cases?

Reply:
Yeah, we likely do. We don't need to test for invalid yaml, because that's the responsibility of the yaml parser. But I'm going to include some tests for yaml that's valid but isn't a valid CFN template